services/serviceutils/src/main/java/tech/pegasys/artemis/service/serviceutils/TypedService.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.service.serviceutils;
+
+import tech.pegasys.artemis.util.async.SafeFuture;
+
+public abstract class TypedService<T> extends Service {
```

Comment:
Looks like we're only using it for testing - I don't think we should be creating new interfaces just to facilitate testing.

Reply:
removed